app/src/main/java/org/simple/clinic/widgets/qrcodescanner/QrCodeScannerView.kt

```diff
@@ -0,0 +1,97 @@
+package org.simple.clinic.widgets.qrcodescanner
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import com.budiyev.android.codescanner.AutoFocusMode
+import com.budiyev.android.codescanner.CodeScanner
+import com.budiyev.android.codescanner.CodeScannerView
+import com.budiyev.android.codescanner.DecodeCallback
+import com.budiyev.android.codescanner.ErrorCallback
+import com.budiyev.android.codescanner.ScanMode
+import com.google.zxing.BarcodeFormat
+import com.jakewharton.rxbinding2.view.RxView
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers.mainThread
+import org.simple.clinic.activity.TheActivity
+import org.simple.clinic.widgets.ScreenCreated
+import org.simple.clinic.widgets.ScreenDestroyed
+import org.simple.clinic.widgets.TheActivityLifecycle
+import org.simple.clinic.widgets.UiEvent
+import timber.log.Timber
+import javax.inject.Inject
+
+class QrCodeScannerView(context: Context, attrs: AttributeSet) : FrameLayout(context, attrs) {
+
+  @Inject
+  lateinit var activity: TheActivity
+
+  @Inject
+  lateinit var lifecycle: Observable<TheActivityLifecycle>
+
+  @Inject
+  lateinit var controller: QrCodeScannerViewController
+
+  private val scannerView = CodeScannerView(context).apply {
+    isAutoFocusButtonVisible = false
+    isFlashButtonVisible = false
+    maskColor = Color.TRANSPARENT
+    frameColor = Color.TRANSPARENT
+  }
+
+  private val codeScanner by lazy(LazyThreadSafetyMode.NONE) { CodeScanner(context, scannerView) }
+
+  @SuppressLint("CheckResult")
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    if (isInEditMode) {
+      return
+    }
+    TheActivity.component.inject(this)
+    addView(scannerView, FrameLayout.LayoutParams.MATCH_PARENT, FrameLayout.LayoutParams.MATCH_PARENT)
```

Comment:
`FrameLayout.LayoutParams.MATCH_PARENT` -> `LayoutParams.MATCH_PARENT` because you're already inside `FrameLayout`?

Reply:
Done.